feat: Update Conversations SDK to 4.0.3

AlejandroOrozco opened the pull request:
<!-- Describe your Pull Request -->

**Context:**
Since the latest version the `type` (ChannelType) property from the Participant is no longer available from Swift context, instead it was replaced for a basic enum for Swift. Previously the `type` was being stored in CoreData as an integer, so the idea behind this update is to stored the channel as a plain String, since it will be use only for display purposes.

**Changes:**
- Upgrade Conversations SDK to 4.0.3
- Add data migration to change `type` for the String version.
- Store channel type.

| Type Marked as Unavailable |  StoredAsInt16 |  Channel Type BasicEnum |
|:--:|:--:|:--:|
|<img width="1325" alt="Marked as Unavaible for Swift" src="https://github.com/twilio/twilio-conversations-demo-ios-swift/assets/4898461/3428daa0-5ebe-47af-8bdf-31db65c59949">|<img width="1084" alt="StoredAsInt16" src="https://github.com/twilio/twilio-conversations-demo-ios-swift/assets/4898461/4c0d3021-1657-4115-81d9-e3ffc591445e">|<img width="683" alt="CreatedAsBasicEnum" src="https://github.com/twilio/twilio-conversations-demo-ios-swift/assets/4898461/71b5a840-9093-4a33-8745-77574bd2fb65">|

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

AlejandroOrozco committed:
feat: Update Conversations SDK to 4.0.2

boy1bot2 commented:
https://patch-diff.githubusercontent.com/raw/twilio/twilio-conversations-demo-ios-swift/pull/22.patch

AlejandroOrozco committed:
feat: Update Conversations SDK to 4.0.3
- Fix UI Updates from background threads.
- Fix Memory Allocation on the ObservableResultPublisher.
- Fix Errors on CoreData fetching.
- Subscribe to conversations on Background thread & release cancellables for The ConversationsManager.

rusmonster commented on ConversationsApp/Models/CoreData/ChannelTypeDataMigration.swift:
```diff
@@ -0,0 +1,59 @@
+                channel = "other"
+            case 2:
+                channel = "chat"
+            case 3:
```
As I see from [this screenshot](https://private-user-images.githubusercontent.com/4898461/286286439-3428daa0-5ebe-47af-8bdf-31db65c59949.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1ODkxOTAsIm5iZiI6MTcxNzU4ODg5MCwicGF0aCI6Ii80ODk4NDYxLzI4NjI4NjQzOS0zNDI4ZGFhMC01ZWJlLTQ3YWYtOGJkZi0zMWRiNjVjNTk5NDkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MDVUMTIwMTMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDE0MGRiMjVlZWQ2NWNjODk2ZmUzZTM4YWY1OTdkY2I2OWE0OWRhZTU4N2JhYmZhNjc5OTFkNjQ4NjgxMDEzYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.IhM_OB0qFUg6jhBNM01iWwHrvvheIY0l-GYSSDD_ifA) is should be like:
```
case 3:
   channel = "sms"
case 4:
   channel = "whatsapp"
```
isn't it?

AlejandroOrozco replied:
I think you're right, added both. Thank you 👍

rusmonster commented on ConversationsApp/Models/CoreData/PersistentParticipantDataItem+CoreDataExt.swift:
```diff
@@ -34,8 +34,9 @@ extension PersistentParticipantDataItem {
-            self.type = Int16(participant.type.rawValue)
-            
+            // TODO: Properly convert this channelType to a string, currently it is just a Swift enum and the channel string is hidden for Swift.
+            self.channel = String(describing: participant.channelType)
```
Why don't use just [Participant.channel](https://sdk.twilio.com/ios/conversations/releases/4.0.3/docs/Classes/TCHParticipant.html#//api/name/channel) here?

AlejandroOrozco replied:
That variable is not visible in the Swift Context, I think the target is missing the bridging-header to automate that translation between Objc & Swift.

![Screenshot 2024-06-13 at 5 57 42 PM](https://github.com/twilio/twilio-conversations-demo-ios-swift/assets/4898461/376d76cf-0450-4e42-93d6-071b03beee56)

AlejandroOrozco committed:
fix: Address Comments
